### PR TITLE
Integration of dense embedding model Snowflake arctic-embed-m-v1.5

### PR DESCRIPTION
**Pull request by Stefan824**
This update integrates the Snowflake arctic-embed-m-v1.5 dense embedding model into the Pyserini toolkit. A new class was developed to support the entire snowflake-series of models, offering a scalable foundation for future expansions, such as additional Snowflake embedding model integrations. This modification is intended to lay the groundwork for smoother integration of future models from the Snowflake series

### CODE REVIEW

**Comment by Stefan824** [comment]
If is linked to resolving issue https://github.com/castorini/pyserini/issues/1959

**Review by lintool — CHANGES_REQUESTED** [review]

File: pyserini/search/faiss/_searcher.py [hunk 1]
```diff
@@ -362,6 +364,43 @@ def encode(self, query: str, **kwargs):
             return self.get_embedding(inputs)
         else:
             return super().encode(query)
+        
+        
+class ArcticQueryEncoder(QueryEncoder):
```
[the remark this answers is not in the document]
Comment (Stefan824) [reply pyserini/search/faiss/_searcher.py:369]: Yeah that is how the current version of pyserini does for all query encoders. Shivani and I were talking about this too. Maybe we could make a separate issue on this matter?

[the remark this answers is not in the document]
Comment (lintool) [reply pyserini/search/faiss/_searcher.py:369]: @MXueguang ping on this?

**Review by lintool — CHANGES_REQUESTED** [review]
lg, but can you please add test cases, like in: https://github.com/castorini/pyserini/blob/master/tests/test_encoder_model_dpr.py

So maybe test case for the document encoder first - add query encoder test cases when we merge in the BEIR regressions?

**Review by lintool — APPROVED** [review]

**Pull request merged**